pipeline/stage02_processing/scripts/spatial_downsampling.py

```diff
@@ -16,9 +16,29 @@
 def spatial_smoothing(images, MACRO_PIXEL_DIM):
 
     # Now we need to reduce the noise from the images by performing a spatial smoothing
-    images = measure.block_reduce(images, (1, MACRO_PIXEL_DIM, MACRO_PIXEL_DIM), np.mean, cval = np.median(images))
+    images_reduced = measure.block_reduce(images, (1, MACRO_PIXEL_DIM, MACRO_PIXEL_DIM), np.mean, cval = np.median(images))
```

Comment:
The images can have empty sites that have nan values. Averaging or taking the median including them also results in a nan value, thus np.mean and np.median should be np.nanmean and np.nanmedian.